src/extensions/__tests__/extension-discovery.test.ts

```diff
@@ -43,6 +43,7 @@ jest.mock("../extension-installer", () => ({
     installPackage: jest.fn(),
   },
 }));
+jest.mock("fs-extra");
```

Comment:
FYI: instead of monkey-patching global file system, we'll soon be able to cleanly achieve the same by overriding dependencies. To the future! :)